### PR TITLE
1.5.0 final work

### PR DESCRIPTION
**Pull request by tacaswell**
Branch I am dumping commits on as I start to do the release.  I am re-writing the release guide as I go.

### CODE REVIEW

File: unit/memleak_hawaii3.py [hunk 1]
```diff
@@ -39,13 +39,23 @@
...
-print('Average memory consumed per loop: %1.4fk bytes\n' % ((end - start) / float(indEnd - indStart)))
+print('Average memory consumed per loop: %1.4fk bytes\n' %
+      ((end - start) / float(indEnd - indStart)))
+fig, ax = plt.subplots()
+ax2 = ax.twinx()
+ax.plot(mem_size, 'r')
+ax.set_ylabel('memory size', color='r')
+ax2.plot(coll_count, 'k')
+ax2.set_ylabel('collect count', color='k')
+fig.savefig('report')
```
Comment (mdboom) [inline unit/memleak_hawaii3.py]: Does this fix the leak, or just provide more info?  In other words, should I try looking into this?

Comment (tacaswell) [inline unit/memleak_hawaii3.py]: Just more information.  I was checking that the memory increase is uniform (which it is :disappointed: )

I don't think it is all in one place either.  Removing all of the plotting calls, and just making, saving, and closing the figures/axes results in a 3k/loop leak.

I am leaning towards this not blocking the release.

Comment (mdboom) [inline unit/memleak_hawaii3.py]: I suppose the real question is how much worse 1.5 is than 1.4.3...  Even then, maybe not blockable -- some magical fix could always go into 1.5.1

Comment (tacaswell) [inline unit/memleak_hawaii3.py]: running in in 1.4.3 now, but it looks just as bad (from eye balling the prints)

Comment (tacaswell) [inline unit/memleak_hawaii3.py]: py3.5, mpl 1.4.3: Average memory consumed per loop: 45.2139k bytes
py2.7, mpl 1.4.3: Average memory consumed per loop: 0.1642k bytes

Comment (mdboom) [inline unit/memleak_hawaii3.py]: Maybe that explains why Python 3 always feels sluggish :imp:

File: doc/users/whats_new.rst [hunk 2]
```diff
@@ -8,7 +8,7 @@ For a list of all of the issues and pull requests since the last
 revision, see the :ref:`github-stats`.
 
 .. note::
-   matplotlib 1.5 supports Python 2.6, 2.7, 3.3, 3.4, and 3.5
+   matplotlib 1.5 supports Python 2.7, 3.4, and 3.5
```
Comment (jkseppan) [inline doc/users/whats_new.rst]: There's a similar line in https://github.com/tacaswell/matplotlib/blob/1.5.0_final_work/INSTALL#L187

File: unit/memleak_hawaii3.py (hunk 1 above)
Comment (efiring) [inline unit/memleak_hawaii3.py]: Python 3 startup time is also much slower than python 2.

File: doc/users/whats_new.rst (hunk 2 above)
Comment (tacaswell) [inline doc/users/whats_new.rst]: Good catch, thanks

File: unit/memleak_hawaii3.py (hunk 1 above)
Comment (efiring) [inline unit/memleak_hawaii3.py]: Somehow, we need to make checking for memory leakage standard, not necessarily on every commit, but at least something that can be triggered weekly.  In my opinion, memory leaks are such fundamental errors that they should have top priority for attention.  Easy for me to say, though; tracking them down is not something for which I have much competence or time.

Comment (tacaswell) [inline unit/memleak_hawaii3.py]: I find the python version dependence strange,  I thought the gc got _better_ in 3...

Comment (WeatherGod) [inline unit/memleak_hawaii3.py]: Probably matters by your definition of "better". Better in keeping memory
under control while not causing much overhead, or better in keeping memory
usage to its smallest amounts?

**Comment by tacaswell** [comment]
Should have the tag up in the next 15-30 min

**Pull request merged**